app/src/main/java/com/github/se/signify/ui/screens/challenge/NewChallengeScreen.kt

```diff
@@ -55,111 +53,78 @@ fun NewChallengeScreen(
-                    .background(MaterialTheme.colorScheme.primary)
-                    .testTag("TopBar"))
-      },
-      content = { padding ->
-        Box(modifier = Modifier.fillMaxSize().padding(padding).testTag("NewChallengeScreen")) {
-          // Back button aligned to the top-left corner
-          BackButton(
-              onClick = { navigationActions.goBack() },
-          )
+  AnnexScreenScaffold(navigationActions = navigationActions, testTagColumn = "NewChallengeScreen") {
+    // My Friends button
+    UtilTextButton(
+        onClickAction = { navigationActions.navigateTo("Friends") },
+        testTag = "MyFriendsButton",
+        text = "My Friends",
+        backgroundColor = MaterialTheme.colorScheme.primary,
+    )
 
-          Column(
-              modifier =
-                  Modifier.fillMaxSize()
-                      .padding(
-                          top = 80.dp,
-                          start = 16.dp,
-                          end = 16.dp) // Padding to avoid overlap with the back button
-                      .testTag("NewChallengeContent"),
-              horizontalAlignment = Alignment.CenterHorizontally) {
-                Spacer(modifier = Modifier.height(70.dp))
+    Spacer(modifier = Modifier.height(32.dp)) // Increased space between buttons
 
-                // My Friends button
-                UtilTextButton(
-                    onClickAction = { navigationActions.navigateTo("Friends") },
-                    testTag = "MyFriendsButton",
-                    text = "My Friends",
-                    backgroundColor = MaterialTheme.colorScheme.primary,
-                )
+    // Create a challenge button
+    UtilTextButton(
+        onClickAction = { navigationActions.navigateTo("CreateChallenge") },
+        testTag = "CreateChallengeButton",
+        text = "Create a Challenge",
+        backgroundColor = MaterialTheme.colorScheme.primary,
+    )
 
-                Spacer(modifier = Modifier.height(40.dp)) // Increased space between buttons
+    Spacer(modifier = Modifier.height(32.dp)) // Increased space between buttons and the box
 
-                // Create a challenge button
-                UtilTextButton(
-                    onClickAction = { navigationActions.navigateTo("CreateChallenge") },
-                    testTag = "CreateChallengeButton",
-                    text = "Create a Challenge",
-                    backgroundColor = MaterialTheme.colorScheme.primary,
-                )
+    // Ongoing Challenges Section
+    Box(
+        modifier =
+            Modifier.fillMaxWidth()
+                .border(2.dp, MaterialTheme.colorScheme.onPrimary)
+                .background(MaterialTheme.colorScheme.primary)
+                .padding(16.dp)
+                .testTag("OngoingChallengesBox")) {
+          Column(
+              horizontalAlignment = Alignment.CenterHorizontally,
+              modifier = Modifier.testTag("OngoingChallengesContent")) {
+                Text(
```

Comment:
UtilTextButton could have been a good choice here

Reply:
This part is not a button, but the a text description for the box.

Reply:
I meant utilText that I tought It was in Utils as it's something common to use.
In that case a feature work would be to add a utilTextBox for exmaple for more modularity